File: Dataset/Leetcode/valid/62/61.java
 class Solution {
    public int XXX(int m, int n) {
        ms = new int[m][n];
        for (int i = 0;i < m;i++) {
            for (int j = 0;j < n;j++) {
                ms[i][j] = -1;
            }
        }
        
        return dfs(m,n,0,0);
    }
    
    int ms[][];
    
    private int dfs(int m,int n,int x,int y) {
        if (ms[x][y] != -1) {
            return ms[x][y];
        }
        
        if (x == m - 1 && y == n - 1) {
            return 1;
        }

        if (x == m - 1 && y < n - 1) {
            return 1;
        }
        
        if (y == n - 1 && x < m - 1) {
            return 1;
        }
        
        if (x + 1 < m && y + 1 < n) {
            return ms[x][y] = dfs(m,n,x + 1,y) + dfs(m,n,x,y + 1);
        }
        
        return 0;
    }
}

File: Dataset/Leetcode/train/56/61.cpp
class Solution {
public:
    vector<vector<int>> XXX(vector<vector<int>>& intervals) {
        if (intervals.empty()) return {};
        sort(intervals.begin(), intervals.end(), [](const vector<int> &a, const vector<int> &b) {
            return a[0] < b[0];
        });
        vector<vector<int> > ans;
        vector<int> last = std::move(intervals[0]);
        const int N = intervals.size();
        for (int i = 1; i < N; ++i) {
            vector<int> &cur = intervals[i];
            if (cur[0] > last[1]) {
                ans.emplace_back(std::move(last));
                last = std::move(cur);
            } else {
                if (cur[1] > last[1]) last[1] = cur[1];
            }
        }
        ans.emplace_back(std::move(last));
        return ans;
    }
};

